test/utilities/extensionUtils.ts

```diff
@@ -35,14 +34,29 @@ export type Extension = {
   packageJSON: unknown;
 };
 
-type ExtensionType = {
+export type ExtensionType = {
   extensionId: ExtensionId;
   name: string;
   vsixPath: string;
   shouldInstall: 'always' | 'never' | 'optional';
   shouldVerifyActivation: boolean;
 };
 
+export type ExtensionActivation = {
+  extensionId: string;
+  isPresent: boolean;
+  version?: string;
+  activationTime?: string;
+  hasBug?: boolean;
+};
+
+export type VerifyExtensionsOptions = {
+  timeout?: number;
+  interval?: number;
+};
+
+const VERIFY_EXTENSIONS_TIMEOUT = 20_000;
```

Comment:
All of these time values that we track are cumbersome, so the general questions is what to do with these and more importantly how do we adjust these without code changes.

Reply:
with an env var or a secret that we pass to the workflow? Like we do for sfdx_auth_url

Reply:
I saw in the pause function that an EnvironmentSettings exposes variable `throttleFactor`, which is only used by pause.

I guess something along those lines, if need be. My hope is that we, via better state transition validation, rely less-and-less on hard pauses.
I know we will not be able to get rid of them completely, but there is hope.

I don't see a THROTTLE_FACTOR value setup in any of the workflows, so we are not using this right now.
I think at a minimum we should find a way to augment this with platform specific throttle values.
This allows us to keep the minimum wait times for the needed pauses, but be able to apply a throttle factor to each platform as necessary.

The other bit that I think is needed